### PR TITLE
[feat] add Hashtable On GPU

### PR DESCRIPTION
**Pull request by rhdong**
- add Hashtable On GPU
- switch cuda tool chain to cuda11
- update STYLE GUIDE for clang format on MacOS.
- update bazel version to 3.7.2

### CODE REVIEW

File: .bazelversion [hunk 1]
```diff
@@ -1 +1 @@
-3.1.0
\ No newline at end of file
+3.7.2
```
Comment (Lifann) [inline .bazelversion:1]: Is the bazel version 3.7.2 necessary? If it is, please also mention it in README.

File: build_deps/toolchains/gcc7_manylinux2010-nvcc-cuda11/BUILD [hunk 2]
```diff
@@ -57,9 +57,9 @@ cc_toolchain_config(
         "/dt7/usr/lib/gcc/x86_64-pc-linux-gnu/7/include",
         "/dt7/usr/lib/gcc/x86_64-pc-linux-gnu/7/include-fixed",
         "/dt7/usr/include",
-        "/usr/local/cuda-10.1/targets/x86_64-linux/include",
-        "/usr/local/cuda-10.1/include",
-        "/usr/local/cuda-10.1/extras/CUPTI/include",
+        "/usr/local/cuda-11.0/targets/x86_64-linux/include",
```
Comment (Lifann) [inline build_deps/toolchains/gcc7_manylinux2010-nvcc-cuda11/BUILD:60]: Would it be possible when someone successfully builds TFRA with cuda version other than 11.0, or specify the cuda version according to the Tensorflow version.

**Review by Lifann — COMMENTED** [review]
I made several review comments. Please check it.

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+    LOG(INFO) << "HashTable on GPU is created successfully:"
+              << " K=" << std::type_index(typeid(K)).name()
+              << ", V=" << std::type_index(typeid(V)).name()
+              << ", max_size=" << max_size_ << ", min_size=" << min_size_;
+  }
+
+  ~CuckooHashTableOfTensorsGpu() { delete table_; }
+
+  void CreateTable(size_t max_size, gpu::TableWrapperBase<K, V>** pptable) {
+    if (runtime_dim_ <= 50) {
```
[the remark this answers is not in the document]
Comment (rhdong) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:90]: Just run one time, and mod50 & switch will have worse readability

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+    int64 default_total = default_flat.size();
+    bool is_full_default = (total == default_total);
+
+    cudaStream_t _stream;
+
+    if (len > 0) {
+      size_t default_value_num =
+          is_full_default ? default_value.shape().dim_size(0) : 1;
+      CUDA_CHECK(cudaStreamCreate(&_stream));
+      CUDA_CHECK(cudaMalloc((void**)&d_status, sizeof(bool) * len));
```
[the remark this answers is not in the document]
Comment (rhdong) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:143]: I have tried but fail, and I will consider you advice next version, for this is a stable implement.

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+                      const Tensor& values) override {
+    size_t len = keys.flat<K>().size();
+    K* d_keys;
+    gpu::ValueArrayBase<V>* d_values;
+    if (len > 0) {
+      cudaStream_t _stream;
+      CUDA_CHECK(cudaStreamCreate(&_stream));
+      CUDA_CHECK(cudaMalloc((void**)&d_keys, sizeof(K) * len));
+      CUDA_CHECK(cudaMalloc((void**)&d_values, sizeof(V) * runtime_dim_ * len));
+      CUDA_CHECK(cudaMemcpy((void*)d_keys, (void*)keys.tensor_data().data(),
```
[the remark this answers is not in the document]
Comment (rhdong) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/cuckoo_hashtable_op_gpu.cu.cc:255]: I tested it, sometimes inputs were in CPU memory that caused crash, so I used memcpy..

**Review by Lifann — APPROVED** [review]
LGTM

**Review by Lifann — APPROVED** [review]

**Pull request merged**